### PR TITLE
Refactoring _get_environment.

### PR DESCRIPTION
**Pull request by dhermes**
- Removes use of "global" by putting the env name in another
  namespace.
- Splits logic into digestible pieces.
- Also makes slight change to GoogleCredentials docstring.

This is the first chunk of #80. I'm happy to wait until #82 is in and rebase.

### CODE REVIEW

File: oauth2client/client.py [hunk 1]
```diff
@@ -952,37 +972,19 @@ class GoogleCredentials(OAuth2Credentials):
   Here is an example of how to use the Application Default Credentials for a
   service that requires authentication:
 
-  <code>
```
Comment (craigcitro) [inline oauth2client/client.py:955]: oh -- were these `<code>` tags in there for epydoc? (_sigh_ epydoc)

File: oauth2client/client.py [hunk 2]
```diff
@@ -952,37 +972,19 @@ class GoogleCredentials(OAuth2Credentials):
...
+      service = build('compute', 'v1', credentials=credentials)
 
-  print(response)
-  </code>
+      PROJECT = 'bamboo-machine-422'
+      ZONE = 'us-central1-a'
+      request = service.instances().list(project=PROJECT, zone=ZONE)
+      response = request.execute()
 
-  A service that does not require authentication does not need credentials
```
Comment (craigcitro) [inline oauth2client/client.py:972]: any reason not to keep this one?

Comment (dhermes) [inline oauth2client/client.py:972]: Yes. It is in a credentials class, so discussing a service that doesn't require credentials is off-topic.

File: oauth2client/client.py (hunk 1 above)
Comment (dhermes) [inline oauth2client/client.py:955]: I don't know anything about epydoc, so it's unclear. I hope not.

Comment (craigcitro) [inline oauth2client/client.py:955]: can you try a quick `make docs` and see how they look?

File: oauth2client/client.py (hunk 2 above)
Comment (craigcitro) [inline oauth2client/client.py:972]: s'fair.

File: oauth2client/client.py (hunk 1 above)
Comment (dhermes) [inline oauth2client/client.py:955]: http://google.github.io/oauth2client/doc/oauth2client.client.GoogleCredentials-class.html
is the place it gets spit out and it puts the code block in a `<pre>` tag, which is a literal environment, so the tag does nothing (as you can see).

On re-build, the content is still in a `<pre>` environment.

**Pull request merged**